apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/http/Http.kt

```diff
@@ -22,6 +22,13 @@ interface HttpBody {
    * This can be called several times
    */
   fun writeTo(bufferedSink: BufferedSink)
+
+  /**
+   * Returns true if this body expects at most one call to writeTo and can be transmitted at most once.
+   * This would be the case when reading the source of data consumes it, for instance when uploading files.
+   */
+  val isOneShot: Boolean
+    get() = false
```

Comment:
I don't like this too much. That one more API that's useful to a small portion of the usecases but accounts for 25% of the interface. It also introduce default methods, which is something I'm never sure how backward compatible it is.

Instead, I'd much rather fail early in `DefaultUpload` if the body has been transmitted already.